lib/view_component/compiler.rb

```diff
@@ -146,21 +146,26 @@ def matching_views_in_source_location
       source_location = component_class.source_location
       return [] unless source_location
 
-      location_without_extension = source_location.chomp(File.extname(source_location))
-
       extensions = ActionView::Template.template_handler_extensions.join(",")
 
-      # view files in the same directory as the component
-      sidecar_files = Dir["#{location_without_extension}.*{#{extensions}}"]
-
       # view files in a directory named like the component
       directory = File.dirname(source_location)
       filename = File.basename(source_location, ".rb")
       component_name = component_class.name.demodulize.underscore
 
+      sub_component_files = if component_class.name.include?("::")
+        subcomponent_path = component_class.name.split("::")[0...-1].map(&:underscore).join("/")
+        Dir["#{directory}/#{subcomponent_path}/#{component_name}.*{#{extensions}}"]
+      else
+        []
+      end
```

Comment:
Can you clarity what is needed here? Not obvious to me what sub_component_files are in this context.

Reply:
I'll try to add a comment to help clarify in the code, but the gist of it is that this adds support for defining a component inside of another component. 

e.g.

```ruby
# frozen_string_literal: true

class SubComponentComponent < ViewComponent::Base
  include ViewComponent::SubComponents

  renders_many :items, -> (highlighted: false) do
    MyHighlightComponent.new(highlighted: highlighted)
  end

  def initialize(classes: "")
    @classes = classes
  end

  class MyHighlightComponent < ViewComponent::Base
    def initialize(highlighted: false)
      @highlighted = highlighted
    end

    def classes
      @highlighted ? "highlighted" : "normal"
    end
  end
end

```

The current behavior is that components nested inside of other components within the same file will assume that the parent component's template is also its template, so defining a template for the nested component doesn't work and adding a `call` method raises.